test/suites/dev-tanssi-relay/proxy/test-proxy-registrar.ts

```diff
@@ -0,0 +1,252 @@
+                await context.createBlock([await tx.signAsync(sudoAlice)]);
+
+                const proxies = await polkadotJs.query.proxy.proxies(sudoAlice.address);
+                expect(proxies.toJSON()[0]).to.deep.equal([
+                    {
+                        delegate: delegateBob.address,
+                        proxyType: "SudoRegistrar",
+                        delay: 0,
+                    },
+                ]);
+            },
+        });
+
+        it({
+            id: "E02",
+            title: "Delegated account can sudo txs in paras_registrar",
+            test: async function () {
+                const PARA_ID = 5555;
+                const txReserve = polkadotJs.tx.proxy.proxy(
+                    sudoAlice.address,
+                    null,
+                    polkadotJs.tx.sudo.sudo(
+                        polkadotJs.tx.registrar.forceRegister(
+                            delegateBob.address,
+                            50,
+                            PARA_ID,
+                            GENESIS_HEAD,
+                            VALIDATION_CODE
+                        )
+                    )
+                );
+                await context.createBlock([await txReserve.signAsync(delegateBob)]);
+
+                const registrar_info = await polkadotJs.query.registrar.paras(PARA_ID);
+                expect(registrar_info.toJSON()).not.toBeNull();
+            },
+        });
+
+        it({
+            id: "E03",
+            title: "Delegated account can sudo txs in data preservers and registrar",
+            test: async function () {
+                // A regular user registers a new avs
+
+                const txReserve = polkadotJs.tx.registrar.reserve();
+                await context.createBlock([await txReserve.signAsync(charlie)]);
+
+                let events = await polkadotJs.query.system.events();
+                const reservedEvent = events.filter((a) => {
+                    return a.event.method == "Reserved" && a.event.data[1].toString() == charlie.address;
+                });
+                const reservedParaId = reservedEvent[0].event.data[0].toPrimitive();
+
+                const txRegisterRelay = polkadotJs.tx.registrar.register(reservedParaId, GENESIS_HEAD, VALIDATION_CODE);
+                await context.createBlock([await txRegisterRelay.signAsync(charlie)]);
+
+                const txRegisterTanssi = polkadotJs.tx.containerRegistrar.register(
+                    reservedParaId,
+                    genesisData,
+                    GENESIS_HEAD
+                );
+                await context.createBlock([await txRegisterTanssi.signAsync(charlie)]);
+
+                await jumpSessions(context, 1);
+
+                // Accept validation code so that para is onboarded after 2 sessions
+                const txAddsCode = polkadotJs.tx.sudo.sudo(
+                    polkadotJs.tx.paras.addTrustedValidationCode(VALIDATION_CODE)
```

Comment:
This call needs also needs sudo but is not supported by this proxy, should we add it @girazoki ?

Reply:
yes

Reply:
Although it is unlikely that we need to use it (nodes to this by default) better add it